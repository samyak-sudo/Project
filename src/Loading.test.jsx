import React from 'react';
import { render, screen } from '@testing-library/react';
import Loading from './Loading';

describe('Loading Component', () => {
    test('renders loading component with correct text', () => {
        render(<Loading />);
        
        // Check if the loading text is present
        const loadingText = screen.getByText('Loading...');
        expect(loadingText).toBeInTheDocument();
        
        // Check if it's inside a div with correct class
        const loadingDiv = loadingText.parentElement;
        expect(loadingDiv).toHaveClass('loading');
    });

    test('matches snapshot', () => {
        const { container } = render(<Loading />);
        expect(container).toMatchSnapshot();
    });

    test('renders h2 element', () => {
        render(<Loading />);
        const headingElement = screen.getByRole('heading', { level: 2 });
        expect(headingElement).toBeInTheDocument();
    });
}); 