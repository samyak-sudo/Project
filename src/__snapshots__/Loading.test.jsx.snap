// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component matches snapshot 1`] = `
<div>
  <div
    class="loading"
  >
    <h2>
      Loading...
    </h2>
  </div>
</div>
`;
